File: kore/src/main/scala/org/kframework/kore/Sort.scala
// Copyright (c) 2014 K Team. All Rights Reserved.

package org.kframework.kore

trait Sort extends SortToString {
  def name: String
}

object Sort {
  def apply(s: String) = UninterpretedSort(s)
  def unapply(s: Sort): Option[String] = Some(s.name)
}

case class UninterpretedSort(name: String) extends Sort

File: kore/src/main/java/org/kframework/kore/mini/Sort.java
package org.kframework.kore.mini;

public class Sort implements org.kframework.kore.Sort {
    private final String name;

    public Sort(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Sort sort = (Sort) o;

        return name.equals(sort.name);

    }

    @Override
    public int hashCode() {
        return name.hashCode();
    }
}
